sources/packages/api/src/database/entities/designation-agreement-location.model.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  Column,
+  Entity,
+  JoinColumn,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+  RelationId,
+} from "typeorm";
+import { DesignationAgreement, InstitutionLocation } from ".";
+import { ColumnNames, TableNames } from "../constant";
+import { RecordDataModel } from "./record.model";
+
+/**
+ * List of locations associated with the designation agreement.
+ * While the designation agreement is in 'Pending' status, this is
+ * the list of locations requested by the Institution to be designated.
+ * Once approved by the Ministry, this list could be changed case the
+ * Ministry do not approve the designation for all location requested
+ * by the institution.
+ */
+@Entity({ name: TableNames.DesignationAgreementLocations })
+export class DesignationAgreementLocation extends RecordDataModel {
+  @PrimaryGeneratedColumn()
+  id: number;
+  /**
+   * Designation agreement id.
+   */
+  @RelationId(
+    (designationAgreementLocation: DesignationAgreementLocation) =>
+      designationAgreementLocation.designationAgreement,
+  )
+  designationAgreementId: number;
+  /**
+   * Designation agreement.
+   */
+  @ManyToOne(
+    () => DesignationAgreement,
+    (designation) => designation.designationAgreementLocations,
+    { eager: false, orphanedRowAction: "delete" },
```

Comment:
i would like to check whether `orphanedRowAction: "delete"` is different from `onDelete:CASCADE`

Reply:
Apparently yes @ann-aot. For this relationship when child is removed from the parent (Designation in this case), the default behavior was try to set "null" to the designation_agreement_id on the child (designations_locations in this case), so it was "removing" the relationship but not the record. Checking the documentation, the orphanedRowAction was added in a version above the one that we were using (that why there is a typeorm version upgrade in this PR).

Reply:
okay got it 👍 Thanks for the explaination